src/data_provider/src/sysInfoWin.cpp

```diff
@@ -207,6 +207,43 @@ class SysInfoProcess final
                 }
             }
 
+            // At least one logical drive couldn't be mapped, trying another use of QueryDosDevice
+            if (logicalDrives.size() != ret.size())
+            {
+                // We avoid using OS_MAXSTR on Windows XP
+                const auto spPhysicalDevices { std::make_unique<char[]>(OS_SIZE_32768) };
+
+                if (QueryDosDevice(nullptr, spPhysicalDevices.get(), OS_SIZE_32768))
+                {
+                    const auto tokens = Utils::splitNullTerminatedStrings(spPhysicalDevices.get());
+                    const auto spDosDevice { std::make_unique<char[]>(OS_SIZE_32768) };
+
+                    for (const auto& token : tokens)
+                    {
+                        // Checking if token is found in logicalDrives to avoid a large map with unnecessary volumes.
+                        // logicalDrives contains a slash at the end but the result of QueryDosDevice doesn't.
+                        if (std::find_if(logicalDrives.begin(), logicalDrives.end(), [&](const auto & logicalDrive)
+                    {
```

Comment:
Could be arrange better the lambda definition here

Reply:
It is indeed hard to read. The Astyle tool forced this strange writing.
I made a commit to fix it.